Implement Second Page of CGCT Demo

MSFTFox opened the pull request:
Implement the second page of the CGCT demo. This demo shows off pattern filling with clipping regions over text as well as an individual attributed string modified for particular ranges.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/microsoft/winobjc-samples/31)
<!-- Reviewable:end -->

DHowett-MSFT commented on Scenarios/CGCT/CGCT/DemoScenarioIntro.m:
```diff
@@ -14,180 +14,210 @@
-
-    CTFrameDraw(frame, context);
-    CGPathRelease(textPath);
+  CGContextSaveGState(context);
```
wait just a sec what happened to this poor file

DHowett-MSFT commented on Scenarios/CGCT/CGCT/DemoScenarioIntro.m:
```diff
@@ -14,180 +14,210 @@
+  CGContextScaleCTM(context, 1.0f, -1.0f);
+  CGContextTranslateCTM(context, bounds.size.width * .25, -bounds.size.height);
+
+  CTFontRef myCFFont = CTFontCreateWithName((__bridge CFStringRef) @"Segoe UI",
```
your "implement page 2" PR changes page 1 _drastically_

DHowett-MSFT commented on Scenarios/CGCT/CGCT/DemoScenarioText.m:
```diff
@@ -0,0 +1,204 @@
+}
+
+- (UIColor *)backgroundColor {
+  return [UIColor colorWithRed:.6 green:.85 blue:1 alpha:1];
```
reformat all of these

DHowett-MSFT commented on Scenarios/CGCT/CGCT/DemoScenarioText.m:
```diff
@@ -0,0 +1,204 @@
+}
+
+static void drawPattern(void *info, CGContextRef context) {
+  UIImage *patternImage = [UIImage
```
did you have to do this manually instead of using UIImage and `UIColor colorWithPatternImage:`?

DHowett-MSFT reviewed (changes requested):
do we have signoff to use those images, or are they PD?

DHowett-MSFT commented:
sooo many format changes

MSFTFox committed:
Implement Second Page of CGCT Demo

Implement the second page of the CGCT demo. This demo
shows off pattern filling with clipping regions over
text as well as an individual attributed string modified
for particular ranges.

MSFTFox committed:
Use self created images.

MSFTFox commented:
The format changes were because I was trying to apply the pre-commit hook to this repo as well. I've done my best to change them back to an acceptable format. As for the images, I've removed those and generated my own patterns.

aballway commented on Scenarios/CGCT/CGCT/DemoScenario.m:
```diff
@@ -17,9 +17,18 @@
-    [NSException raise:NSInvalidArgumentException format:@"Scenario %@ must override -name", self];
+
+-(NSString *)name {
+    [NSException raise : NSInvalidArgumentException format : @"Scenario %@ must override -name", self];
```
nit: clang format these

aballway commented on Scenarios/CGCT/CGCT/DemoScenarioIntro.m:
```diff
@@ -14,18 +14,22 @@
 
 @implementation DemoScenarioIntro
-- (NSString*)name {
+- (NSString *)name {
```
nit: clang format **all** of these

aballway commented on Scenarios/CGCT/CGCT/DemoScenarioText.m:
```diff
@@ -0,0 +1,163 @@
+    CTParagraphStyleRef paragraphRef = CTParagraphStyleCreate(settings, 1);
+
+    NSString *textBox = @"This is my attributed string. There are many like it, "
+        @"but this one is mine.\n\nMy font is my best friend. "
```
👍

aballway reviewed (approved):
Approval pending format

MSFTFox committed:
Format

DHowett-MSFT commented on Scenarios/CGCT/CGCT/DemoScenarioText.m:
```diff
@@ -0,0 +1,182 @@
+}
+
+static void drawPattern(void* info, CGContextRef context) {
+    UIImage* patternImage =
```
I had a question on this commit about the use of `+[UIColor colorWithPatternImage:]`; it appears to have been consumed by fire.

MSFTFox replied:
Yup. This is using the CGPattern callbacks directly. It seems like more of a sample to showcase how to do manual pattern callbacks in CoreGraphics instead of through UIColor (albeit that would do the same thing under the hood).
